serveradmin/servershell/static/js/servershell/autocomplete/search.js

```diff
@@ -14,85 +14,111 @@ $(document).ready(function () {
+                    let new_term = _build_value(request.term, to_complete, attribute, filter_name);
 
+                    choices.push({
+                        'label': `Filter: ${filter[0]}`,
+                        'value': new_term + '(',
+                    });
+                });
+
+                // Autocomplete attribute value if wanted
+                // @TODO support nested values e.g. "project=Any(Reg("
                 if (servershell.search_settings.autocomplete_values) {
+                    // Add attribute values matching
                     let settings = {
+                        'async': false,
                         'data': {
-                            'attribute': last_attribute,
-                            'value': last_value,
+                            'attribute': attribute,
+                            'value': value,
                         },
-                        'async': false,
                         'success': function (data) {
-                            data.autocomplete.forEach(function (attr_value) {
+                            data.autocomplete.forEach(function(match) {
                                 choices.push({
-                                    'label': `AttrVal: ${attr_value}`,
-                                    'value': _build_value(request.term, cur_term, last_attribute, attr_value),
+                                    'label': `AttrVal: ${match}`,
+                                    'value': _build_value(request.term, to_complete, attribute, match),
                                 });
                             });
                         }
                     };
-                    $.ajax(url, settings);
+                    $.ajax(autocomplete_search_url, settings);
                 }
-
-                // Add filter to autocomplete ...
-                // @TODO: Add autocomplete for nested filter values
-                let filters;
-                if (last_value) {
-                    filters = servershell.filters.filter(filter => filter[0].startsWith(last_value));
-                } else {
-                    filters = servershell.filters;
-                }
-                filters.forEach(function (filter) {
-                    choices.push({
-                        'label': `Filter: ${filter[0]}`,
-                        'value': _build_value(request.term, cur_term, last_attribute, filter[0]) + '(',
-                    });
-                });
             }
 
-            // Attributes available
+            // Autocomplete attributes
             let attributes = servershell.attributes.filter(
-                attr => attr.attribute_id.startsWith(cur_term) && attr.type !== 'reverse'
+                a => a.attribute_id.startsWith(to_complete) && a.type !== 'reverse'
             );
-            attributes.slice(0, limit).forEach(function (attribute) {
+            attributes.slice(0, limit).forEach(function(a) {
```

Comment:
If you want it shorter, I would suggest to use `attr` as a compromise. `attr` is short and still common to read as short for `attribute`. Makes it just easier to read. If you can't go for the compromise for some reason, then ok.

Reply:
I just didn't like to read attribute.attribute_id and from the context I think it is visible what it is but I don't have any strong opinion on that either so I'll just revert it.

Reply:
I see your point, however then the cause is the redundant `attribute` in `attribute_id`, changing the parameter name to `a` is causing that while roughly reading the code, I don't know what `a` is anymore and need to read more code to understand what it can be. `a` could stand for more things now. A developer who is used to the code would for sure directly understand it most probably and with `attr` almost every new developer will understand it right away. ;)

Reply:
Okay I will change it to attr if you prefer that.

Reply:
Yes, if we can afford the 3 extra characters (and I mean it really without sarcasm), I would prefer that.

Reply:
Ask the clients browser ;-)

Reply:
We can always minimise our js files if we are taking the client browser into account. ;)